language/tools/move-mv-llvm-compiler/tests/test_common.rs

```diff
@@ -790,3 +793,111 @@ pub fn clean_results<P: AsRef<Path>>(source: P) -> std::io::Result<()> {
 
     Ok(())
 }
+
+pub fn switch_last_two_extensions_and_rename(path: &Path) {
+    if let Some(filename) = path.file_name() {
+        if let Some(filename_str) = filename.to_str() {
+            let mut parts: Vec<&str> = filename_str.split('.').collect();
+
+            if parts.len() >= 3 {
+                // Swap the last two extensions
+                let last_ext = parts.pop().unwrap();
+                let second_last_ext = parts.pop().unwrap();
+                parts.push(last_ext);
+                parts.push(second_last_ext);
+
+                let new_filename = parts.join(".");
+
+                let mut new_path = path.to_path_buf();
+                new_path.set_file_name(new_filename);
+
+                fs::rename(path, new_path).expect("Error in renameing");
+            }
+        }
+    }
+}
+
+pub fn list_files_with_extension(
+    directory_path: &Path,
+    extension: &str,
+) -> std::io::Result<Vec<String>> {
+    let dir = fs::read_dir(directory_path)?;
+
+    let file_names: Vec<String> = dir
+        .filter_map(|entry| {
+            let entry = entry.ok()?;
+            let path = entry.path();
+            if path.is_file() {
+                if let Some(ext) = path.extension() {
+                    if ext == extension {
+                        Some(path.to_string_lossy().to_string())
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    Ok(file_names)
+}
+
+pub fn clean_directory(directory_path: &Path) -> std::io::Result<()> {
```

Comment:
Function `fs::remove_dir_all` will fail if the directory does not exist.
In CI run this call is not important and may be dropped, since the path (with extension "stored_results") does not originally exist.
But it is convenient to have this clean up for repetitious local runs.

Reply:
> Function `fs::remove_dir_all` will fail if the directory does not exist.

Consider ignoring the error if validating the removal is not required for your use case.

Reply:
done